File: 11-container-with-most-water/container-with-most-water.ts
function maxArea(height: number[]): number {
    let left: number =0;
    let right: number = height.length-1;
    let maxArea : number =0;
    
    while(left<right){
        let h:number = Math.min(height[left], height[right]);
        let width:number = right- left;
        maxArea = Math.max(maxArea, h*width);

        if(height[left]< height[right]){
           left++; 
        }
        else{
            right--;
        }
    }
    return maxArea;
};

File: 11-container-with-most-water/container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        left = 0
        right = len(height)-1
        maxArea = 0

        while left< right:
            h = min(height[left], height[right])
            width = right-left
            maxArea = max(maxArea, h*width)

            if height[left]< height[right]:
                left+=1
            else:
                right-=1

        return maxArea   
            
        

        